### PR TITLE
Module/qc/1.0

### PR DESCRIPTION
**Pull request by Kdreval**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [X] I used the cookiecutter template and updated the placeholder rules.

- [X] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules). 

  - [X] All references to the `rules` object (_e.g._ for input files) are wrapped with `str()`.

- [X] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [X] I updated the final target rule (`*_all`) to include every output rule.

- [X] I tested the module on real data for all supported `seq_type` values.

- [X] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [X] I did not set any global wildcard constraints. Any/all wildcard constraints are set on a per-rule basis. 

- [X] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository).

- [X] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [X] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### If applicable

- [ ] I added more granular output subdirectories.

- [ ] I added rules to the `reference_files` workflow to generate any new reference files.

- [ ] I added subdirectories with large intermediate files to the list of `scratch_subdirectories` in the `default.yaml` configuration file.

- [ ] I updated the list of available wildcards for the input files in the `default.yaml` configuration file.

## Checklist for Updated Module

Important! If you are updating the module version, ensure the previous version of the module is restored from master.
If you want to restore a deleted file or directory from the remote master, you can use `git checkout origin/master path/to/file`,
then a `git commit` will ensure that file is tracked on your branch again.
Example:
```
mv modules/strelka/1.1 modules/strelka/1.2
git checkout origin/master modules/strelka/1.1
```

### CODE REVIEW

File: modules/qc/1.0/qc.smk [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+
+# Subset contigs in bed to those present in the reference and sort the output
+# This is needed to pass GATK validation
+rule _qc_sort_baits:
+    input:
+        baits = _qc_get_baits,
+        fai = str(rules._qc_input_references.output.genome_fai)
+    output:
+        baits = CFG["dirs"]["inputs"] + "references/{genome_build}/{baits_regions}.bed",
+        inermediate_baits = temp(CFG["dirs"]["inputs"] + "references/{genome_build}/{baits_regions}.INTEMEDIATE.bed")
```
Comment (lkhilton) [inline modules/qc/1.0/qc.smk]: You have `intermediate` spelled variously as `inermediate` or `INTEMEDIATE`

File: modules/qc/1.0/qc.smk [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+
+def _qc_get_baits(wildcards):
+    CFG = config["lcr-modules"]["qc"]
+    # simplify config options by re-using the capture space bertween e.g grch37 and hs37d5
+    if "38" in str(wildcards.genome_build):
+        this_genome_build = "hg38"
+    else:
+        this_genome_build = "grch37"
+
+    if "baits_regions" in  CFG:
```
Comment (lkhilton) [inline modules/qc/1.0/qc.smk:195]: Any interest in making in an option to specify separate baits and regions files? Picard does output different metrics for baits vs. regions if different files are given.

File: modules/qc/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Changelog
+
+All notable changes to the `qc` module will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [1.0] - 2022-05-12
```
Comment (lkhilton) [inline modules/qc/CHANGELOG.md]: any notes to add here?

**Review by lkhilton — COMMENTED** [review]
Couple of comments/questions. Does this module make the buggy picard_qc module obsolete?

File: modules/qc/1.0/qc.smk (hunk 1 above)
Comment (Kdreval) [reply modules/qc/1.0/qc.smk]: ah. That's right. That's what you get for trying to blind type. My first module where I tried blind typing skills! :sweat_smile:

File: modules/qc/1.0/qc.smk (hunk 2 above)
Comment (Kdreval) [reply modules/qc/1.0/qc.smk:195]: certainly can be added! Not supported in current implementation but definitely an option for the future update! I envision this to be also an option for the samtools stats, it is possible to restrict to target regions there

File: modules/qc/CHANGELOG.md (hunk 3 above)
Comment (Kdreval) [reply modules/qc/CHANGELOG.md]: true, will update!

**Review by Kdreval — COMMENTED** [review]
Thanks @lkhilton ! Yes this does make the picard module obsolete!

**Comment by Kdreval** [comment]
The typos and changelog has been now fixed!

**Review by lkhilton — APPROVED** [review]

**Review by rdmorin — APPROVED** [review]
Nice addition.

**Pull request merged**